### PR TITLE
Add Directions + Building Highlights + Shuttle Schedule + Testing

### PR DESCRIPTION
**Pull request by MichelRahme**
#11 S-4 As a user, I want to be able to switch the language between English and French.
#20 S-2: As a user, I want to be able to see campus building information when I tap over Concordia buildings. 
#22 S-3.2: As a user, I want to be able to obtain directions to my destination starting at my current location.
#24 S-3.4: As a user, I want to be able to choose a method of transportation among given options.
#27 S-6: As a user, I want to be able to use the application in colourblind mode.
#70 Create Automated Tests.
#71 Create Acceptance Test for Sprint 2.

### CODE REVIEW

File: lib/services/directions_service.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'network.dart';
```
Comment (andrelegault) [inline lib/services/directions_service.dart]: We use absolute imports project-wide so this should be an absolute import (per https://fuchsia.dev/fuchsia-src/development/languages/dart/style#do_import_all_files_within_a_package_using_relative_paths)

File: lib/services/network.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  static Future getData(String url) async {
+    print('Calling uri: $url');
+    Response response = await get(url);
+    if (response.statusCode == 200) {
+      return response.body;
+    } else {
+      print(response.statusCode);
+    }
+  }
+}
```
Comment (andrelegault) [inline lib/services/network.dart]: Consider converting to singleton in the future

File: lib/widgets/buildingModels/building_list.dart [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import 'building_information.dart';
```
Comment (andrelegault) [inline lib/widgets/buildingModels/building_list.dart]: Convert to absolute import

File: lib/widgets/buildingModels/building_list.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import 'dart:async' show Future;
```
Comment (andrelegault) [inline lib/widgets/buildingModels/building_list.dart]: Place this import first per https://dart.dev/guides/language/effective-dart/style#do-place-dart-imports-before-other-imports

File: lib/widgets/map_widget.dart [hunk 5]
```diff
@@ -1,85 +1,89 @@
...
-import 'package:flutter/foundation.dart';
+import 'package:concordia_navigation/models/user_location.dart';
+import 'package:concordia_navigation/providers/buildings_data.dart';
+import 'package:concordia_navigation/providers/map_data.dart';
+import 'package:concordia_navigation/services/size_config.dart';
+import 'package:concordia_navigation/services/location_service.dart';
+import 'package:concordia_navigation/widgets/floating_map_button.dart';
 import 'package:flutter/material.dart';
 import 'package:google_maps_flutter/google_maps_flutter.dart';
 import 'dart:async';
```
Comment (andrelegault) [inline lib/widgets/map_widget.dart]: Place first per https://dart.dev/guides/language/effective-dart/style#do-place-dart-imports-before-other-imports

File: lib/widgets/map_widget.dart [hunk 6]
```diff
@@ -1,85 +1,89 @@
...
-const double CAMERA_BEARING = 30;
-const LatLng SGW = LatLng(45.495944, -73.578075);
-const LatLng LOYOLA = LatLng(45.4582, -73.6405);
-bool _campus = true;
-
-//*****UNCOMMENT BELLOW FOR DARK MAP*****
-//String _mapStyle;
+import 'building_widgets/building_marker.dart';
+import 'buildingModels/building_list.dart';
+import 'buildingModels/building_information.dart';
```
Comment (andrelegault) [inline lib/widgets/map_widget.dart]: Convert to absolute import

File: lib/widgets/shuttle_widget.dart [hunk 7]
```diff
@@ -0,0 +1,100 @@
+import 'package:flutter/material.dart';
+import 'package:concordia_navigation/storage/app_constants.dart';
+import 'package:concordia_navigation/services/size_config.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'shuttle_tile.dart';
```
Comment (andrelegault) [inline lib/widgets/shuttle_widget.dart]: Convert to absolute import

File: lib/models/building.dart [hunk 8]
```diff
@@ -1,14 +1,11 @@
-import 'package:flutter/material.dart';
+import 'package:concordia_navigation/storage/app_constants.dart';
```
Comment (andrelegault) [inline lib/models/building.dart]: Consider adding a synonym to make it clear in code that a constant is from that file, i.e., `import 'package:concordia_navigation/storage/app_constants.dart' as constants;`

File: lib/models/itinerary.dart [hunk 9]
```diff
@@ -0,0 +1,51 @@
+import 'package:concordia_navigation/services/directions_service.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'dart:convert';
```
Comment (andrelegault) [inline lib/models/itinerary.dart]: Place first

File: lib/services/network.dart (hunk 2 above)
Comment (MichelRahme) [reply lib/services/network.dart]: Not possible as it's a package.

File: lib/services/directions_service.dart (hunk 1 above)
Comment (MichelRahme) [reply lib/services/directions_service.dart]: Not possible as it's a package.

**Pull request merged**